src/controllers/userController.ts

```diff
@@ -0,0 +1,57 @@
+import User from "@/models/userModel";
+import { Request, Response } from "express";
+import jwt from "jsonwebtoken";
+import { ObjectId } from "mongoose";
+import expressAsyncHandler from "express-async-handler";
+
+export const registerUser = expressAsyncHandler(
+  async (req: Request, res: Response) => {
+    const { name, email, password } = req.body;
+
+    const user: any = await User.create({
+      name,
+      email,
+      password: password,
+    });
+    res.status(201).json({
+      _id: user._id,
+      name: user.name,
+      email: user.email,
+      token: generateJwt(user._id),
+    });
+  }
+);
+
+export const loginUser = expressAsyncHandler(
+  async (req: Request, res: Response) => {
+    const { email, password } = req.body;
+    const user: any = await User.findOne(
+      { email },
+      { password: 1, name: 1, email: 1, role: 1 }
+    );
+    if (!user) {
+      res.status(400);
```

Comment:
shouldn't it be like this
```suggestion
      return res.status(400).send("User Not found");
```

Reply:
As i am using the error handler so the message inside the Error is handled by that error handler . Other wise there will be Error for again setting of  headers as i cannot use return here because of return type issue